### ISSUE TITLE
Application crashes on incorrect input.

### ISSUE THREAD

**Issue by ViralTaco** [ViralTaco/pillz#2]
Current Behavior: 
Entering any string that isn't a correct "DrugID" will cause an assertion failure. 

Expected behavior: 
Show error "invalid selection", keep running, don't save any intermittent state. 

Proposed solution: 
Propose to add a category for the invalid input. 
Restrict the input to ASCII, or UTF-16 input (ie: fixed length. utf8 is a can of worms that I rather leave closed.)
Don't limit the size of the actual buffer, instead "virtually" cap it. ie: allocate enough memory to accommodate any kind of input, as far as the OS will allow, then use at most the first `LONGEST_VALID_STRING_SIZE` (so far undefined) as the "input" passed to the "database". 
The idea is to prevent any unintentional buffer overflow exploit. (By allocating enough memory for the input, even the part we'll discard.) 
An exception should be thrown when a predetermined part of the buffer is being written to, that exception shall be caught, the memory shall be zeroed byte-by-byte, no as-if allowed, then freed, before the program "gracefully" clears the "screen" (stdout), to display an error message, before exiting with failure. 

TL;DR: The main requirement of such a failure is to properly free the memory and not to leak anything. 
Meaning the failure will be "slow". We don't care. It's an invalid use of the app. 
What not to do is leak everything, then crash, maybe saving the "non-deterministic" (as in not thread-safe) state, or not.
That's just bad.
